indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -388,20 +395,21 @@ public void unlock(final Task task, final Interval interval)
   }
 
   /**
-   * Release all locks for a task. Does nothing if the task is not currently locked.
+   * Release all locks for a task and remove task from set of active tasks. Does nothing if the task is not currently locked or not an active task.
    *
    * @param task task to unlock
    */
-  public void unlock(final Task task)
+  public void remove(final Task task)
   {
     giant.lock();
-
     try {
-      for(final TaskLockPosse taskLockPosse : findLockPossesForTask(task)) {
-        unlock(task, taskLockPosse.getTaskLock().getInterval());
-      }
+      log.info("Removing task[%s] from activeTasks", task.getId());
+      for (final TaskLockPosse taskLockPosse : findLockPossesForTask(task)) {
+          unlock(task, taskLockPosse.getTaskLock().getInterval());
+        }
     }
     finally {
+      activeTasks.remove(task.getId());
```

Comment:
unlock should have a finally block of its own, even if this remove is in its own try{}finally{} block.